plugins/modules/proxysql_query_rules_fast_routing.py

```diff
@@ -0,0 +1,470 @@
+        if not check_mode:
+            result['rules'] = self.get_rule_config(cursor)
+            result['changed'], result['rows_affected'] = self.delete_rule_config(cursor)
+            result['msg'] = "Deleted rule from mysql_query_rules_fast_routing"
+            self.manage_config(cursor, result['changed'])
+        else:
+            result['changed'] = True
+            result['msg'] = (
+                "Rule would have been deleted from "
+                "mysql_query_rules_fast_routing, "
+                "however check_mode is enabled."
+            )
+
+# ===========================================
+# Module execution.
+#
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            login_user=dict(default=None, type='str'),
+            login_password=dict(default=None, no_log=True, type='str'),
+            login_host=dict(default="127.0.0.1"),
+            login_unix_socket=dict(default=None),
+            login_port=dict(default=6032, type='int'),
+            config_file=dict(default="", type='path'),
+            username=dict(required=True, type='str'),
+            schemaname=dict(required=True, type='str'),
+            destination_hostgroup=dict(required=True, type='int'),
+            flagIN=dict(default=0, type='int'),
+            comment=dict(default='', type='str'),
+            state=dict(default='present', choices=['present',
+                                                   'absent']),
+            force_delete=dict(default=False, type='bool'),
+            save_to_disk=dict(default=True, type='bool'),
+            load_to_runtime=dict(default=True, type='bool')
+        ),
+        supports_check_mode=True
+    )
+
+    perform_checks(module)
+
+    login_user = module.params["login_user"]
+    login_password = module.params["login_password"]
+    config_file = module.params["config_file"]
+
+    cursor = None
+    try:
+        cursor, db_conn = mysql_connect(
+            module,
+            login_user,
+            login_password,
+            config_file,
+            cursor_class='DictCursor'
+        )
+    except mysql_driver.Error as e:
+        module.fail_json(
+            msg="unable to connect to ProxySQL Admin Module.. %s" % to_native(e)
+        )
+
+    proxysql_query_rule_fast_routing = ProxyQueryRuleFastRouting(module)
+    result = {}
+
+    result['state'] = proxysql_query_rule_fast_routing.state
+
+    if proxysql_query_rule_fast_routing.state == "present":
+        try:
+            if not proxysql_query_rule_fast_routing.check_rule_cfg_exists(cursor):
+                if proxysql_query_rule_fast_routing.config_data["username"] and \
+                   proxysql_query_rule_fast_routing.config_data["schemaname"] and \
+                   proxysql_query_rule_fast_routing.config_data["flagIN"] and \
+                   proxysql_query_rule_fast_routing.check_rule_pk_exists(cursor):
+                    proxysql_query_rule_fast_routing.update_rule(
```

Comment:
```suggestion
                    proxysql_query_rule_fast_routing.update_rule(
```
does this compare the current state and desired or just checks that it exists and updates all the time?